es6-shim.js

```diff
@@ -2097,6 +2097,88 @@
     // Shim incomplete iterator implementations.
     addIterator(Object.getPrototypeOf((new globals.Map()).keys()));
     addIterator(Object.getPrototypeOf((new globals.Set()).keys()));
+
+    // Reflect
+    if (!globals.Reflect) {
+      globals.Reflect = {
+
+        // Syntax in a functional form.
+        get: function (target, key, receiver) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          return target[key];
+        },
+
+        set: function (target, key, value, receiver) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          target[key] = value;
+        },
+
+        has: function (target, key) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          return key in target;
+        },
+
+        deleteProperty: function (target, key) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          delete target[key];
+        },
+
+        enumerate: function (target) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          var keys = [];
+
+          for (var key in target) {
+            keys.push(key);
+          }
+
+          return new ArrayIterator(keys, 'value');
```

Comment:
It was just an example to show that the hole should be ignored, and not the full implementation :)